jsonschema/src/keywords/one_of.rs

```diff
@@ -22,16 +27,51 @@ impl OneOfValidator {
         context: &CompilationContext,
     ) -> CompilationResult<'a> {
         if let Value::Array(items) = schema {
+            let draft = context.config.draft();
```

Comment:
I wonder if some of this could move to the `config`.. at least the `discriminated_mapping` and `discriminator_property_name`